### PR TITLE
BUG: fix  AttributeError: CompileError object has no attribute orig

### LINKED ISSUE #50062: BUG: AttributeError: CompileError object has no attribute orig
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
df = pd.DataFrame(...)
df.to_sql(...)
```

### Issue Description

Function pandas.DataFrame.to_sql() writes records stored in a DataFrame to a SQL database. Databases supported by SQLAlchemy are supported. This function will call 'sql.py' in 'pandas/pandas/io/sql.py'.

When applied to a database supported by SQLAlchemy, 'sql.py' will use the class 'SQLAlchemyEngine(BaseEngine)', which has this line causing the reported bug: https://github.com/pandas-dev/pandas/blob/main/pandas/io/sql.py#L1392.

Basically, method 'insert_records' handles 'SQLAlchemyError' exceptions. But not all child class of base class 'SQLAlchemyError' has the attribute 'orig'. Doc: https://docs.sqlalchemy.org/en/14/core/exceptions.html

Therefore, adding a checking attribute step will fix this bug. 

### Expected Behavior

AttributeError: CompileError object has no attribute orig 

### Installed Versions

<details>

'1.5.2'

</details>

### PR DESCRIPTION
**Pull request by thanhbui221**
- [x] closes #50062 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v1.5.2.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by thanhbui221** [comment]
Pandas version: 1.5.2.
AttributeError: CompileError object has no attribute orig

File: pandas/io/sql.py [hunk 1]
```diff
@@ -1389,9 +1389,11 @@ def insert_records(
             # https://stackoverflow.com/a/67358288/6067848
             msg = r"""(\(1054, "Unknown column 'inf(e0)?' in 'field list'"\))(?#
             )|inf can not be used with MySQL"""
-            err_text = str(err.orig)
-            if re.search(msg, err_text):
-                raise ValueError("inf cannot be used with MySQL") from err
+            if hasattr(err, 'orig'):
```
Comment (nguyenmphu) [inline pandas/io/sql.py:1392]:
```suggestion
            if hasattr(err, "orig"):
```

**Comment by thanhbui221** [comment]
Function pandas.DataFrame.to_sql() writes records stored in a DataFrame to a SQL database. Databases supported by SQLAlchemy are supported. This function will call 'sql.py' in 'pandas/pandas/io/sql.py'.

When applied to a database supported by SQLAlchemy, 'sql.py' will use the class 'SQLAlchemyEngine(BaseEngine)', which has this line causing the reported bug: https://github.com/pandas-dev/pandas/blob/main/pandas/io/sql.py#L1392.

Basically, method 'insert_records' handles 'SQLAlchemyError' exceptions. But not all child class of base class 'SQLAlchemyError' has the attribute 'orig'. Doc: https://docs.sqlalchemy.org/en/14/core/exceptions.html

Therefore, adding a checking attribute step will fix this bug.

**Review by phofl — CHANGES_REQUESTED** [review]
Hi, 1.5.2 is already out, so this has to go to 1.5.3.

Also, please add tests

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.
